Python/pystate.c

```diff
@@ -2933,6 +2933,29 @@ _str_shared(PyThreadState *tstate, PyObject *obj,
     return 0;
 }
 
+static PyObject *
+_new_float_object(_PyCrossInterpreterData *data)
+{
+    double * value_ptr = data->data;
+    return PyFloat_FromDouble(*value_ptr);
+}
+
+static int
+_float_shared(PyThreadState *tstate, PyObject *obj,
+             _PyCrossInterpreterData *data)
+{
+    if (_PyCrossInterpreterData_InitWithSize(
+            data, tstate->interp, sizeof(double), obj,
```

Comment:
As with `int`, we don't need to keep the object around, since we pass it as intrinsic data.

```suggestion
            data, tstate->interp, sizeof(double), NULL,
```